main.cpp

```diff
@@ -1,14 +1,26 @@
-#include <iostream>
 #include "i.h"
 #include "u.h"
+#include <iostream>
 
 using namespace std;
 using i32 = mgt::i<32>;
+using u32 = mgt::u<32>;
 
 int main() {
   i32 a, b;
-  cin >> a >> b;
-  cout << (a < b) << endl;
+  bool flag = true;
+  for (int i = -1000; i <= 1e3; ++i) {
+    for (int j = i; j <= 1e3; ++j) {
+      a = i, b = j;
+      if (a * b != i32(i * j)) {
+        cout << "Error when i = " << i << ", j = " << j << ", expected " << i32(i * j) << ", got " << a * b << ". "
```

Comment:
many these code should be put into `test.cpp` rather than main

Reply:
I'll take this advice...though testing whether the code works or not in main.cpp is exactly what has been done before.